sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -193,6 +193,10 @@ case class CreateDataSourceTableAsSelectCommand(
             sessionState.catalog.lookupRelation(tableIdent)) match {
             case l @ LogicalRelation(_: InsertableRelation | _: HadoopFsRelation, _, _) =>
               existingSchema = Some(l.schema)
+            case s: SimpleCatalogRelation
+                if s.metadata.properties.contains("spark.sql.sources.provider") =>
```

Comment:
might be good to add a method `CatalogTable.isDatasourceTable` to avoid duplicating this config